lib/runner.js

```diff
@@ -0,0 +1,117 @@
+/*eslint no-console: 0*/
+
+'use strict';
+
+var configLoader = require('./config-loader');
+var Lesshint = require('./lesshint');
+var Vow = require('vow');
+
+var EXIT_CODES = {
+    OK: 0,
+    WARNING: 1,
+    ERROR: 2,
+    NOINPUT: 66,
+    SOFTWARE: 70,
+    CONFIG: 78
+};
+
+var Runner = function (settings) {
+    this.settings = settings;
+};
+
+Runner.prototype.run = function () {
+    var lesshint = new Lesshint();
+    var exitDefer = Vow.defer();
+    var exitPromise = exitDefer.promise();
+    var promises = [];
+    var config;
+
+    if (!this.settings.args.length) {
+        console.error('No files to lint were passed. See lesshint -h');
```

Comment:
Wouldn't it be better to reject the promise with a error message and a error code and keep all the `console.log`s in `cli.js`. So this file can be used from another Node program without any logging output?